tests/MSBuildDeviceIntegration/Tests/InstallAndRunTests.cs

```diff
@@ -88,5 +90,539 @@ public void SubscribeToAppDomainUnhandledException ()
+			}
+
+			// [Test] TryCreateInstanceOfXmlPreservedLinkerClass
+			try {
+				var asm = typeof (Library1.SomeClass).Assembly;
+				var o = Activator.CreateInstance (asm.GetType (""Library1.LinkerClass""));
+				Android.Util.Log.Info (TAG, $""[PASS] Able to create instance of '{o.GetType ().Name}'."");
+			} catch (Exception ex) {
+				Android.Util.Log.Info (TAG, $""[FAIL] Unable to create instance of 'LinkerClass'.\n{ex}"");
+			}
+
+			// [Test] TryAccessXmlPreservedMethodOfLinkerClass
+			try {
+				var asm = typeof (Library1.SomeClass).Assembly;
+				var t = asm.GetType (""Library1.LinkerClass"");
+				var m = t.GetMethod (""WasThisMethodPreserved"");
+				Android.Util.Log.Info (TAG, $""[PASS] Able to locate method '{m.Name}'."");
+			} catch (Exception ex) {
+				Android.Util.Log.Info (TAG, $""[FAIL] Unable to access 'WasThisMethodPreserved ()' method of 'LinkerClass'.\n{ex}"");
+			}
+
+			// [Test] TryAccessAttributePreservedMethodOfLinkerClass
+			try {
+				var asm = typeof (Library1.SomeClass).Assembly;
+				var t = asm.GetType (""Library1.LinkerClass"");
+				var m = t.GetMethod (""PreserveAttribMethod"");
+				Android.Util.Log.Info (TAG, $""[PASS] Able to locate method '{m.Name}'."");
+			} catch (Exception ex) {
+				Android.Util.Log.Info (TAG, $""[FAIL] Unable to access 'PreserveAttribMethod ()' method of 'LinkerClass'.\n{ex}"");
+			}
+
+			// [Test] TryAccessXmlPreservedFieldOfLinkerClass
+			try {
+				var asm = typeof (Library1.SomeClass).Assembly;
+				var t = asm.GetType (""Library1.LinkerClass"");
+				var m = t.GetProperty (""IsPreserved"");
+				Android.Util.Log.Info (TAG, $""[PASS] Able to locate field '{m.Name}'."");
+			} catch (Exception ex) {
+				Android.Util.Log.Info (TAG, $""[FAIL] Unable to access 'IsPreserved' field of 'LinkerClass'.\n{ex}"");
+			}
+
+			// [Test] TryCreateInstanceOfNonXmlPreservedClass
+			try
+			{
+				var asm = typeof (Library1.SomeClass).Assembly;
+				var o = Activator.CreateInstance (asm.GetType (""Library1.NonPreserved""));
+				Android.Util.Log.Info (TAG, $""[LINKALLFAIL] Able to create instance of '{o.GetType ().Name}' which should have been linked away."");
+			} catch (Exception ex) {
+				Android.Util.Log.Info (TAG, $""[LINKALLPASS] Unable to create instance of 'NonPreserved' as expected.\n{ex}"");
+			}
+
+			// [Test] TryAccessNonXmlPreservedMethodOfLinkerModeFullClass
+			var newLinkerTestInstance = new CustomLinkerDescriptionTests ();
+			Android.Util.Log.Info (TAG, newLinkerTestInstance.TryAccessNonXmlPreservedMethodOfLinkerModeFullClass ());
+
+			Android.Util.Log.Info (TAG, Library2.Bug21578.MulticastOption_ShouldNotBeStripped ());
+			Android.Util.Log.Info (TAG, Library2.Bug21578.MulticastOption_ShouldNotBeStripped2 ());
+			Android.Util.Log.Info (TAG, Library2.Bug35195.AttemptCreateTable ());
+			Android.Util.Log.Info (TAG, Library2.Bug36250.SerializeSearchRequestWithDictionary ());
+
+			Android.Util.Log.Info (TAG, ""All regression tests completed."");
+");
```

Comment:
Since the above is a massive string, should we make this an `EmbeddedResource` in its own `.cs` file, so you can read the file more easily? It wouldn't have intellisense, but at least you don't have to escape `""`.

You could load it similar to:
https://github.com/xamarin/xamarin-android/blob/e4405eb27e740c93cd90f993f13734309c013193/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Android/XamarinFormsAndroidApplicationProject.cs#L20-L21

Reply:
Same idea with the other large strings here. Maybe you would end up with 4 separate files?

The `.xml` file is short enough, it's probably fine.

Reply:
I've moved the four larger test files to @(EmbeddedResource) files.